pkg/controller/schedulingmanager/controller.go

```diff
@@ -0,0 +1,176 @@
+	c := &SchedulerController{
+		scheduler:        make(map[string]schedulingtypes.Scheduler),
+		config:           config,
+		fedNamespace:     fedNamespace,
+		clusterNamespace: clusterNamespace,
+		targetNamespace:  targetNamespace,
+		minimizeLatency:  minimizeLatency,
+		stopChan:         stopChan,
+	}
+
+	c.worker = util.NewReconcileWorker(c.reconcile, util.WorkerTiming{})
+
+	c.store, c.controller = cache.NewInformer(
+		&cache.ListWatch{
+			// Only watch the federation namespace to ensure
+			// restrictive authz can be applied to a namespaced
+			// control plane.
+			ListFunc: func(options metav1.ListOptions) (pkgruntime.Object, error) {
+				return client.FederatedTypeConfigs(fedNamespace).List(options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return client.FederatedTypeConfigs(fedNamespace).Watch(options)
+			},
+		},
+		&corev1a1.FederatedTypeConfig{},
+		util.NoResyncPeriod,
+		util.NewTriggerOnAllChanges(c.worker.EnqueueObject),
+	)
+
+	return c, nil
+}
+
+// Run runs the Controller.
+func (c *SchedulerController) Run(stopChan <-chan struct{}) {
+	go c.controller.Run(stopChan)
+
+	// wait for the caches to synchronize before starting the worker
+	if !cache.WaitForCacheSync(stopChan, c.controller.HasSynced) {
+		runtime.HandleError(fmt.Errorf("Timed out waiting for cache to sync"))
+		return
+	}
+
+	c.worker.Run(stopChan)
+}
+
+func (c *SchedulerController) reconcile(qualifiedName util.QualifiedName) util.ReconciliationStatus {
+	key := qualifiedName.String()
+
+	glog.Infof("Running reconcile FederatedTypeConfig for %q", key)
+
+	cachedObj, exist, err := c.store.GetByKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("Failed to query FederatedTypeConfig store for %q: %v", key, err))
+		return util.StatusError
+	}
+	if !exist {
+		return util.StatusAllOK
+	}
+	typeConfig := cachedObj.(*corev1a1.FederatedTypeConfig)
+
+	deleted := typeConfig.DeletionTimestamp != nil
+	if deleted {
+		// Do not support deletion yet
+		return util.StatusAllOK
+	}
+
+	// set name and group for the type config target
+	corev1a1.SetFederatedTypeConfigDefaults(typeConfig)
+	apiResource := typeConfig.GetTarget()
+	templateKind := typeConfig.GetTemplate().Kind
+
+	for kind, schedulingType := range schedulingtypes.SchedulingTypes() {
+		for _, fedKind := range SchedulingRegistry[kind] {
+			if fedKind == templateKind {
+				// Scheduling preference controller is started on demand
+				_, ok := c.scheduler[kind]
+				if !ok {
+					scheduler, err := schedulingpreference.StartSchedulingPreferenceController(kind, schedulingType.SchedulerFactory, c.config, c.fedNamespace, c.clusterNamespace, c.targetNamespace, c.stopChan, util.DefaultClusterAvailableDelay, util.DefaultClusterUnavailableDelay, c.minimizeLatency)
```

Comment:
As per a previous comment, `clusterAvailableDelay` and `clusterUnavailableDelay` should use passed-in values rather than being defaulted.